doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,88 @@
+
+In our initial implementation, we will only support extracting the top level archive to a temporary location. This means that the `NestedInstallerFile` must not be contained in any nested archives. The appropriate install flow will proceed based on the specified `NestedInstallerType`. 
+
+> If the community presents enough use cases that require decompressing additional layers, then we will consider extending this functionality and adding a separate manifest entry to override the default behavior of only unzipping the top level archive.
+
+> Since we are utilizing Windows Shell APIs, we need to ensure that we are not invoking a UI during the extraction. This can be done by [setting the operation flag to not display any UI](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-ifileoperation-setoperationflags). 
+
+> During implementation, we will also need to ensure that this process can work under SYSTEM context.
+
+## ZIP Threat Detection
+ZIP and other archive file types are known threat vectors for malware in the form of ZIP compression bombs. Two of the most common types of compression bombs are multi-layered (recursive) and single-layered (non-recursive). ZIP bombs rely on a repetition of identical files that have extremely large compression ratios. Some examples include the following:
+
+1. **Multi-layered (recursive)**:
+A ZIP file containing multiple layers of nested ZIP files that achieve high compression ratios when decompressed recursively. This technique is often used to bypass compression ratio checks performed by ZIP parsers for a single layer.
+
+2. **Single-Layered (non-recursive)**
+A ZIP bomb that expands fully after a single round of decompression. The extremely high compression ratio of the ZIP bomb is achieved by overlapping files within the ZIP container. 
+
+In order to protect our users from these possible threats, we will need to scan the ZIP file for malware using [Pure, a static analysis file format checker](https://github.com/ronomon/pure). 
+
+> Pure is licensed under the [MIT license](https://github.com/ronomon/pure/blob/master/LICENSE).
+
+If the Pure library functions that are called on a given ZIP file detect malware, the process will terminate and a warning will be displayed to the user. The user can include `--override-archive-malware-checks` to bypass this check and continue installation at their own risk.
+
+These checks will only applied in the following scenarios to minimize unnecessary costs on performance during installation. 

```

Comment:
```suggestion
These checks will only applied when any of the following conditions are met to minimize unnecessary costs on performance during installation. 
```
Clarification that these are meant to be ‘or’ and not ‘and’

Reply:
only *be* applied?

Reply:
Changed to suggested